core/src/main/java/feast/core/model/FeatureStatistics.java

```diff
@@ -0,0 +1,269 @@
+    CommonStatistics commonStatistics =
+        CommonStatistics.newBuilder()
+            .setNumNonMissing(count - numMissing)
+            .setNumMissing(numMissing)
+            .setMaxNumValues(maxNumValues)
+            .setMinNumValues(minNumValues)
+            .setTotNumValues(totalNumValues)
+            .setNumValuesHistogram(Histogram.parseFrom(numValuesHistogram))
+            .build();
+
+    switch (featureNameStatisticsBuilder.getType()) {
+      case INT:
+      case FLOAT:
+        NumericStatistics numStats =
+            NumericStatistics.newBuilder()
+                .setCommonStats(commonStatistics)
+                .setMean(mean)
+                .setStdDev(stdev)
+                .setNumZeros(zeroes)
+                .setMin(min)
+                .setMax(max)
+                .setMedian(median)
+                .addHistograms(Histogram.parseFrom(numericValueHistogram))
+                .addHistograms(Histogram.parseFrom(numericValueQuantiles))
+                .build();
+        featureNameStatisticsBuilder.setNumStats(numStats);
+        break;
+      case STRING:
+        StringStatistics.Builder stringStats =
+            StringStatistics.newBuilder()
+                .setCommonStats(commonStatistics)
+                .setUnique(unique)
+                .setAvgLength(averageLength);
+        if (rankHistogram == null) {
+          stringStats.setRankHistogram(RankHistogram.getDefaultInstance());
+        } else {
+          stringStats.setRankHistogram(RankHistogram.parseFrom(rankHistogram));
+        }
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(topValues)) {
+          ObjectInputStream ois = new ObjectInputStream(bis);
+          List<StringStatistics.FreqAndValue> freqAndValueList =
+              (List<StringStatistics.FreqAndValue>) ois.readObject();
+          stringStats.addAllTopValues(freqAndValueList);
+        } catch (IOException | ClassNotFoundException e) {
+          throw new InvalidProtocolBufferException(
+              "Failed to parse field: StringStatistics.TopValues. Check if the value is malformed.");
+        }
+        featureNameStatisticsBuilder.setStringStats(stringStats);
+        break;
+      case BYTES:
+        BytesStatistics bytesStats =
+            BytesStatistics.newBuilder()
+                .setCommonStats(commonStatistics)
+                .setAvgNumBytes(avgBytes)
+                .setMinNumBytes(minBytes)
+                .setMaxNumBytes(maxBytes)
+                .build();
+        featureNameStatisticsBuilder.setBytesStats(bytesStats);
+        break;
+      case STRUCT:
+        StructStatistics structStats =
+            StructStatistics.newBuilder().setCommonStats(commonStatistics).build();
+        featureNameStatisticsBuilder.setStructStats(structStats);
+        break;
+    }
+    return featureNameStatisticsBuilder.build();
+  }
+
+  private static FeatureStatistics fromProto(FeatureNameStatistics featureNameStatistics)
```

Comment:
It's confusing that this overlaps with `fromProtos` above. It could have been a non-static method like `updateFromFeatureNameStatistics`